tests/lib/rules/jsx-no-literals.js

```diff
@@ -284,6 +284,10 @@ ruleTester.run('jsx-no-literals', rule, {
         <img alt='blank image'></img>
       `,
     },
+    {
+      code: '<div>&mdash;</div>',
+      options: [{ noStrings: true, allowedStrings: ['&mdash;'] }],
+    },
```

Comment:
let's add a few more test cases covering the behavior of `—` in the jsx and `'&mdash;'` in the config, as well as the reverse.

Reply:
Will do! Good idea. I'll update the PR in the next day or so.

Reply:
@joeattardi any update?

Reply:
Sorry for the lack of responsiveness on this.

I was working on adding some more tests and I'm not sure what's going on.

I added the following test to the `valid` section:

```
    {
      code: '<div>&mdash;</div>',
      only: true,
      options: [{ noStrings: true, allowedStrings: ['—'] }],
    },
```

When I run the tests, this fails:

```
 AssertionError [ERR_ASSERTION]: Should have no errors but had 1: [
  {
    ruleId: 'jsx-no-literals',
    severity: 1,
    message: 'Strings not allowed in JSX files: "&mdash;"',
    line: 1,
    column: 6,
    nodeType: 'Literal',
    messageId: 'noStringsInJSX',
    endLine: 1,
    endColumn: 13
  }
]
```

However if I add the exact same test to the `invalid` section, with the expected error:

```
{
      code: '<div>&mdash;</div>',
      only: true,
      options: [{ noStrings: true, allowedStrings: ['—'] }],
      errors: [
        {
          messageId: 'noStringsInJSX',
          data: { text: '&mdash;' },
        },
      ],
    },
```

This test fails, too:

```
AssertionError [ERR_ASSERTION]: Should have 1 error but had 0: []
```

Maybe I'm missing something about how the tests work?

Reply:
On which parser? It’s possible different parsers produce different results.